### DISCUSSION TITLE
Bug or Feature? WFS beeing converted to gpkg when uploading project to qfield cloud

### DISCUSSION THREAD

**Discussion by Spacedatists** [opengisch/QField#5307] [Q&A]
No matter what I do, my WFS is always beeing converted into a geopackage when uploading my project to qfield cloud. No problem with that when exporting to an offline project via cable export. Any Idea on that?

**Comment by romanmeyerblch** [comment]
seconded, still not working

**Comment by nirvn** [comment]
@Spacedatists , @romanmeyerblch , did you properly set your WFS layer action to "Directly access source" in the project properties' QField tab:

![image](https://github.com/user-attachments/assets/725aadfc-d60e-44f3-b8e6-2d16a0a53013)

?

**Reply by romanmeyerblch** [reply]
Yes, twice. 
1. In the qgis-Project before creating the qgiscloud-project. 
2. in the qgiscloud-project before synchronising to the cloud.

gpkg gets created in the first step, and there's no way I can avoid it.
QGIS 3.34.14, QField Sync 4.13.0

**Reply by romanmeyerblch** [reply]
Workaround for now: 
1. create qgis-project with wfs- and other layers, set wfs to 'directly access the data source'.
2. sync to cloud, qfieldcloud-project gets created, and a gpkg for each wfs-layer
3. remove the wfs-gpkg-layers, delete gpkgs in the cloud-folder 
4. add wfs layers. Again. 
5. sync project to cloud and explicitly delete the wfs-gpkgs in the process
6. tadaa, you can now work with wfs-layers in qfield!

all jokes aside, I don't understand why wfs-layers get converted/exported to gpkg when creating the qfieldcloud-project, even though I explicitly select 'directly access the data source'. I also tried a mix of offline/online wfs-layers, but they ALL get converted to gpkg.

**Reply by nirvn** [reply]
@romanmeyerblch , ohhhh, you are using the convert currently opened project option right? That will indeed force your datasets into gpkg (as per the description). So yes, you should not add WFS layers prior to converting a project to a cloud project.

Alternatively, you can create an empty cloud project from scratch and skip the conversion. If you do that, you have to manually copy all of your file-based datasets into that newly-created cloud project's folder.

**Reply by romanmeyerblch** [reply]
Thanks for the explanation @nirvn. Doesn't make much sense to me as a user. My understanding was: create your qgis-project the way you need it, then convert to cloud and use it in the field. I wasn't expecting having to fiddle again with the project at this point. I understood from the description that shape and other formats would be converted to gpkg. I wouldn't have dreamed that a wfs gets converted!
